pkg/sql/opt/testutils/testcat/test_catalog.go

```diff
@@ -98,11 +103,26 @@ func (tc *Catalog) ResolveDataSourceByID(
 // resolveDataSource checks if `toResolve` exists among the data sources in this
 // Catalog. If it does, resolveDataSource updates `name` to match `toResolve`,
 // and returns the corresponding data source. Otherwise, it returns an error.
-func (tc *Catalog) resolveDataSource(toResolve, name *tree.TableName) (opt.DataSource, error) {
+// If createVTable is true, if the table isn't found, we will attempt to lazily
+// resolve and add a virtual table with the given name.
+func (tc *Catalog) resolveDataSource(
+	toResolve, name *tree.TableName, createVTable bool,
```

Comment:
I'm not following why we need a `createVTable` parameter. I'd expect to *first* try to resolve the name as a vtable, and only if we can't find a matching vtable do we fall back on `tc.dataSources`. I'd expect to do this every time. Can you show a concrete example where that wouldn't work?